### DISCUSSION TITLE
privacy edition image

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#56] [Ideas]
Some ideas for a privacy enhanced image

# Bundled plugins

https://github.com/HelloChatterbox/dummy_wakeword_plugin - disable wake word
https://github.com/JarbasLingua/jarbas-stt-plugin-vosk - offline STT (make default ? )

# Extra software

https://github.com/OpenJarbas/mock-backend - no phoning home

# Patches (waiting for mycroft....)

https://github.com/MycroftAI/mycroft-core/pull/2698 - skill settings GUI

# Default config

TODO 

- point to mock backend (localhost)
- use vosk for offline STT
- set flag to [disable skill settings sync](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L130)
- disable [opt-in](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L39)
- disable [skill_manifest upload](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L108)
- disable [wake_word upload](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L167) and point to [dummy url](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L168)
- disable [auto skills update](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L111)
- set [update interval](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L120) to an absurd huge number just in case  (defense against future bugs in msm/core)
- blacklist https://github.com/JarbasSkills/skill-mock-backend because backend is running system wide
- blacklist https://github.com/MycroftAI/skill-pairing
- blacklist https://github.com/MycroftAI/skill-configuration

# System tweaks

- firewall blocking messagebus
- firewall blocking GUI messagebus
- firewall blocking mock-backend

# For discussion
- blacklist https://github.com/MycroftAI/skill-installer , the voice search sucks and its tied to mycroft marketplace, might end up with accidental skill installs
	- there is https://github.com/OpenVoiceOS/ovos_skill_manager in the making
	- alternative companion skill is in TODO list, not locked to appstores and with GUI planned
- replace wolfram alpha and weather with alternative that dont use mycroft backend
	- there is https://github.com/JarbasSkills/skill-wolfie but i would have to make a weather alternative
- what [models](https://alphacephei.com/vosk/models) should we include for vosk/offline stt ?
- change [messagebus port](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L136) to specifically avoid connection attempts
- change [GUI port](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/configuration/mycroft.conf#L136) to specifically avoid connection attempts, NOTE: not sure this can be done, does the mycrof-gui also allow changing this?
- bundle hivemind to encourage proper secure bus connections, so people dont allow mycroft bus and are instead pointed to this
- blacklist https://github.com/JarbasHiveMind/skill-hivemind because hivemind master is running system wide 
- can we add some warning when users make firewall changes? specifically is allowing any of the blocked ports?

**Comment by JarbasAl** [comment]
disabling msm is now possible as follows

```
  "skills": {
    "msm": {
      "repo": {
        "url": "https://github.com/OpenVoiceOS/disable-msm-dummy-repo",
        "branch": "master"
      }
    }
```

**Comment by NeonDaniel** [comment]
Couple additional skills adapted to remove backend dependency:

Weather: https://github.com/NeonGeckoCom/skill-weather
Stock: https://github.com/NeonGeckoCom/skill-stock
